woocommerce/woocommerce-ios
# [Product Multi-selection] Resolve toggle state for product multi-selection

Closes: #8920

## Description:

This PR addresses the toggle state (selected, and unselected) for `Product` and `Product variations` when product multi-selection is enabled. 

At the moment it is expected that unselecting a product **does not** remove the product from the order, this will be dealt with on a future PR. Both selecting, or unselecting, currently adds a product to the Order.

## Changes:
- `selectProduct` and `selectVariation` are the methods that are called on each view model when we tap on a Product or a Product Variation. We've duplicated the existing `toggleSelection(_:)` logic to account for the closure being called when multi-selection is enabled.
- Renamed `toggleSelectionForVariations` to `toggleSelectionForAllVariations` to reflect that tapping on that toggle will select all variations.
- Removed the call for `toggleSelectionForAllVariations` from the `ProductSelectorView`, as we do not want to select all product variations when the button is tapped.
- Adapted existing Unit Tests to account for the feature flag.
- Added additional Unit Tests to confirm the `productRow` selected state (`.selected` vs `.unselected`) is correct when the feature flag is enabled.

## Testing instructions
- Go to Orders > `+` >  `+ Add Product`
- Add Products to the order. See how the toggle/checkbox is enabled/disabled when tapping
- Go to a variable product and tap on the row. See how you can also select/unselect variable products
- Go back to the Order. Tap on Done.
- Confirm that all products are part of the Order.

| Variable products selection  | Products selection  | Order creation  |
|---|---|---|
| <img width=450 src="https://user-images.githubusercontent.com/3812076/220635012-2e917b31-678b-4743-be91-feef2ced3537.png"> | <img width=450 src="https://user-images.githubusercontent.com/3812076/220635052-c5d7a276-1b8d-44f7-8fe8-37a3992cd1f9.png"> | <img width=450 src="https://user-images.githubusercontent.com/3812076/220635107-76208151-8031-46e3-bf7e-db94150de7cb.png"> |
